packages/11ty/content/_assets/javascript/application/accordion.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Accordion class
+ * @param {HTMLElement} element Accordion section element
+ */
+const Accordion = class {
+  constructor(element) {
+    this.copyLinkButton = element.querySelector('.accordion-section__copy-link-button')
+    this.element = element
+    this.id = element.getAttribute('id')
+    this.globalExpand = document.querySelector('.global-accordion-expand-all')
+    this.globalCollapse = document.querySelector('.global-accordion-collapse-all')
+  }
+
+  /**
+   * @return {NodeList} All of this accordion's parent accordions, if accordion is nested
+   */
+  get parentAccordions() {
+    return Array.from(Accordion.elements).filter(
+      (el) => el.contains(this.element) && el != this.element
+    )
+  }
+
+  static className = 'accordion-section'
+  static globalExpand = document.querySelector('.global-accordion-expand-all')
+  static globalCollapse = document.querySelector('.global-accordion-collapse-all')
+  static elements = document.querySelectorAll(`.${this.className}`)
+  // Static method to check if an element is or is included in an accordion
+  static partOfAccordion = (element) => element.closest(`.${this.className}`)
+
+  static setGlobalControls() {
+    if (Accordion.elements.length === 0) return
+    if (!Accordion.globalCollapse || !Accordion.globalExpand) return;
+    const closedCount = Array.from(Accordion.elements).filter((element => element.getAttribute('open') === null)).length
+    if (closedCount !== 0 && closedCount < Accordion.elements.length) {
+      Accordion.globalCollapse.classList.remove('visually-hidden')
+      Accordion.globalExpand.classList.remove('visually-hidden')
+    }
+    if (closedCount === Accordion.elements.length) {
+      Accordion.globalExpand.classList.remove('visually-hidden')
+      Accordion.globalCollapse.classList.add('visually-hidden')
+    }
+    if (closedCount === 0 && Accordion.elements.length > 0) {
+      Accordion.globalExpand.classList.add('visually-hidden')
+      Accordion.globalCollapse.classList.remove('visually-hidden')
+    }
```

Comment:
A small state machine may be easier to reason about and maintain; the states might be `allClosed`, `allOpen`, and perhaps `noControls` or `displayFalse`.

Reply:
I didn't add a state machine per se but I renamed some things that hopefully make this easier to follow.